spec/models/api_request_spec.rb

```diff
@@ -28,25 +28,64 @@
       end
     end
 
-    describe "without combinations of host, path, and query" do
+    describe "without combinations of host, path, and parameters" do
       it "returns an error for missing host" do
-        # mock in path and query
+        # mock in path and parameters
         allow_any_instance_of(ApiRequest).to receive(:path).and_return "/path"
-        allow_any_instance_of(ApiRequest).to receive(:query).and_return "?some=query"
+        allow_any_instance_of(ApiRequest).to receive(:parameters).and_return []
         expect { ApiRequest.new.response }.to raise_error(RuntimeError)
       end
       it "returns an error for missing path" do
-        # mock in path and query
+        # mock in path and parameters
         allow_any_instance_of(ApiRequest).to receive(:host).and_return "example.com"
-        allow_any_instance_of(ApiRequest).to receive(:query).and_return "?some=query"
+        allow_any_instance_of(ApiRequest).to receive(:parameters).and_return []
         expect { ApiRequest.new.response }.to raise_error(RuntimeError)
       end
       it "returns an error for missing query" do
-        # mock in path and query
+        # mock in path and parameters
         allow_any_instance_of(ApiRequest).to receive(:host).and_return "example.com"
         allow_any_instance_of(ApiRequest).to receive(:path).and_return "/path"
         expect { ApiRequest.new.response }.to raise_error(RuntimeError)
       end
     end
+
+    context "with differing pagination requests" do
+      let(:api) { ApiRequest.new }
+
+      before do
+        success = Net::HTTPSuccess.new(1.0, "200", "OK")
+        allow_any_instance_of(Net::HTTP)
+          .to receive(:start)
+          .and_return(success)
+        expect(success)
+          .to receive(:body)
+          .and_return('{"meta":{},"data":[]}')
+
+        # mock in host, path, and query
+        allow_any_instance_of(ApiRequest).to receive(:host).and_return "example.com"
+        allow_any_instance_of(ApiRequest).to receive(:path).and_return "/path"
+        allow_any_instance_of(ApiRequest).to receive(:parameters).and_return []
+      end
+
+      it "returns no offset if no page is set" do
+        api.response
+        uri = api.instance_variable_get(:@uri)
```

Comment:
[thought] hmm this feels like we are testing private variables, which we shouldn't be doing.

instead we should be checking to make sure that `URI::HTTPS.build` and  `Net::HTTP.start` are being called with the correct arguments.

`expect(URI::HTTPS).to receive(:build).with("example.com", "/path", "username=#{@username}").and_return $some_mocked_object`

Reply:
(this is pseduo code, happy to work with you on it if this doesn't work quite right, i always have to look up how to do this)

Reply:
I tried that, because that was your suggestion the other day when I asked, and ended up banging my head against an annoying amount of things before I gave up and went with `@uri` instead.  I don't feel great about it, though.  I think if we need this PR in right away, then someone else may have to pick up fixing these tests because I can't spend a lot of time on it today, but otherwise if next week would work I would definitely revisit it with you!

Reply:
[suggestion] I agree with testing via expectations like Davida said. If that won't work, for whatever reason, then I'd rather pull the `request_uri` method to public and test that rather than reaching in to the instance variable.

Reply:
I think I've done it!  By jove.  I figured out what I was doing wrong, too, so that's great news for my future